api/test/integration/indicator-record/indicator-records.service.spec.ts

```diff
@@ -0,0 +1,364 @@
+      MaterialsToH3sService,
+    );
+  });
+
+  afterEach(async () => {
+    //Delete all generated data
+    await indicatorRecordRepository.delete({});
+    await indicatorRepository.delete({});
+    await sourcingRecordRepository.delete({});
+    await adminRegionRepository.delete({});
+    await supplierRepository.delete({});
+    await businessUnitRepository.delete({});
+    await h3DataRepository.delete({});
+    await geoRegionRepository.delete({});
+    await materialRepository.delete({});
+
+    await deleteH3GridTables();
+  });
+
+  describe('createIndicatorRecordsBySourcingRecords', () => {
+    test('When creating Indicator Records providing indicator coefficients, it should create the records properly', async () => {
+      // ARRANGE
+      const indicatorPreconditions = await createPreconditions();
+
+      const sourcingData = {
+        sourcingRecordId: indicatorPreconditions.sourcingRecord1.id,
+        tonnage: indicatorPreconditions.sourcingRecord1.tonnage,
+        geoRegionId: indicatorPreconditions.sourcingLocation1.geoRegionId,
+        materialId: indicatorPreconditions.sourcingLocation1.materialId,
+      };
+
+      const providedCoefficients: IndicatorCoefficientsDto = {
+        [INDICATOR_TYPES.BIODIVERSITY_LOSS]: 0.1,
+        [INDICATOR_TYPES.CARBON_EMISSIONS]: 0.4,
+        [INDICATOR_TYPES.DEFORESTATION]: 0.35,
+        [INDICATOR_TYPES.UNSUSTAINABLE_WATER_USE]: 0.2,
+      };
+
+      //ACT
+      await indicatorRecordService.createIndicatorRecordsBySourcingRecords(
+        sourcingData,
+        providedCoefficients,
+      );
+
+      //ASSERT
+      const allIndicators = await indicatorRecordRepository.find();
+      expect(allIndicators.length).toEqual(4);
+
+      //Value is provided coeff * tonnage, and scaler is null, because it's based on production coefficient, and it's not provided
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.DEFORESTATION,
+        indicatorPreconditions.deforestation,
+        sourcingData,
+        350,
+        null,
+      );
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.BIODIVERSITY_LOSS,
+        indicatorPreconditions.biodiversityLoss,
+        sourcingData,
+        100,
+        null,
+      );
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.CARBON_EMISSIONS,
+        indicatorPreconditions.carbonEmissions,
+        sourcingData,
+        400,
+        null,
+      );
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.UNSUSTAINABLE_WATER_USE,
+        indicatorPreconditions.waterRisk,
+        sourcingData,
+        200,
+        null,
+      );
+    });
+
+    it("When creating indicator record with no provided coefficients, and there's no H3 data for the given material, it should throw an error it should create corresponding CANCELLED Sourcing Locations, with the appropiate Sourcing Records and Indicator Records", async () => {
```

Comment:
Very minor one: Can we keep using test() through all the project? Just for consistency